aspnetcore/client-side/spa-services.md

```diff
@@ -0,0 +1,323 @@
+
+## Creating a new project
+
+JavaScriptServices provides pre-configured application templates. SpaServices is used in these templates, in conjunction with different frameworks and libraries such as Angular, Aurelia, Knockout, React, and Vue.
+
+These templates can be installed via the .NET Core CLI by running the following command:
+
+```console
+dotnet new --install Microsoft.AspNetCore.SpaTemplates::*
+```
+
+A list of available SPA templates is displayed:
+
+| Templates                                 | Short Name | Language | Tags        |
+|:------------------------------------------|:-----------|:---------|:------------|
+| MVC ASP.NET Core with Angular             | angular    | [C#]     | Web/MVC/SPA |
+| MVC ASP.NET Core with Aurelia             | aurelia    | [C#]     | Web/MVC/SPA |
+| MVC ASP.NET Core with Knockout.js         | knockout   | [C#]     | Web/MVC/SPA |
+| MVC ASP.NET Core with React.js            | react      | [C#]     | Web/MVC/SPA |
+| MVC ASP.NET Core with React.js and Redux  | reactredux | [C#]     | Web/MVC/SPA |
+| MVC ASP.NET Core with Vue.js              | vue        | [C#]     | Web/MVC/SPA | 
+
+To create a new project using one of the SPA templates, include the **Short Name** of the template in the `dotnet new` command. The following command creates an Angular application with ASP.NET Core MVC configured for the server-side:
+
+```console
+dotnet new angular
+```
+
+<a name="runtime-config-mode"></a>
+
+### Set the runtime configuration mode
+
+Two primary runtime configuration modes exist:
+1. **Development**:
+    * Includes source maps to ease debugging.
+    * Doesn't optimize the client-side code for performance.
+1. **Production**:
+    * Excludes source maps.
+    * Optimizes the client-side code via bundling & minification.
+
+ASP.NET Core uses an environment variable named `ASPNETCORE_ENVIRONMENT` to store the configuration mode. See **[Setting the environment](xref:fundamentals/environments#setting-the-environment)** for more information.
+
+### Running with .NET Core CLI
+
+Restore the required NuGet and npm packages by running the following command at the project root:
+
+```console
+dotnet restore && npm i
+```
+
+Build and run the application:
+
+```console
+dotnet run
+```
+
+The application starts on localhost according to the [runtime configuration mode](#runtime-config-mode). Navigating to `http://localhost:5000` in the browser displays the landing page.
+
+### Running with Visual Studio 2017
+
+Open the *.csproj* file generated by the `dotnet new` command. The required NuGet and npm packages are restored automatically upon project open. This restoration process may take up to a few minutes; and, the application is ready to run when it completes. Click the green run button or press `Ctrl` + `F5`, and the browser opens to the application's landing page. The application runs on localhost according to the [runtime configuration mode](#runtime-config-mode). 
```

Comment:
`Ctrl` + `F5` --> `Ctrl+F5` or just Ctrl+F5 (so long as style is consistent throughout the doc; we haven't established a preference for one or the other yet)